### PR TITLE
[WIP] start supporting for TypeScript 1.6.0

### PR DESCRIPTION
**Pull request by vvakame**
https://github.com/Microsoft/TypeScript/issues/4081

@basarat @DanielRosenwasser @RyanCavanaugh

### CODE REVIEW

File: async/asyncamd-tests.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/// <reference path="./async.d.ts" />
```
Comment (DanielRosenwasser) [inline async/asyncamd-tests.ts]: Were these just always erroring in TS 1.5 as well?

**Comment by DanielRosenwasser** [comment]
@vvakame LGTM :+1:. Just as a heads up, I'll be sending out more 1.6 fixes soon.

**Comment by DanielRosenwasser** [comment]
@vvakame I have some some of the changes that will fix the broken tests in this PR. I'll send them out to help unblock this.

File: async/asyncamd-tests.ts (hunk 1 above)
Comment (vvakame) [inline async/asyncamd-tests.ts]: no, I think this is breaking change in TS 1.6.
I think 1.6 is correct.
`import async = require("async");` should not lookup from `"./async"`.

```
$  cat a.js
require("b").hello();
$ cat  b.js
exports.hello = function() { console.log("hi!"); };
$ node a.js
module.js:334
    throw err;
          ^
Error: Cannot find module 'b'
    at Function.Module._resolveFilename (module.js:332:15)
    at Function.Module._load (module.js:282:25)
    at Module.require (module.js:361:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/private/tmp/test/a.js:1:63)
    at Module._compile (module.js:426:26)
    at Object.Module._extensions..js (module.js:444:10)
    at Module.load (module.js:351:32)
    at Function.Module._load (module.js:306:12)
    at Function.Module.runMain (module.js:467:10)
```

**Comment by vvakame** [comment]
@DanielRosenwasser cool! :clap:

**Comment by DanielRosenwasser** [comment]
@vvakame try merging from `master` now.

**Comment by vvakame** [comment]
I'm making some dirty fix.

```
$ ag "NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590"
angular2/angular2.d.ts
1370:    // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
1445:        // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590

breeze/breeze-tests.ts
446:    /* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
764:    /* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
795:    /* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590

devextreme/dx.devextreme.d.ts
3079:        // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590

i18next/i18next.d.ts
22:    // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
76:    // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590

react/react-addons-tests.ts
60:        // NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590

sequelize/sequelize-tests.ts
590:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
644:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
726:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
734:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
742:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
748:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
933:/* NOTE https://github.com/borisyankov/DefinitelyTyped/pull/5590
```

**Comment by vvakame** [comment]
great thanks for @DanielRosenwasser !

**Comment by DanielRosenwasser** [comment]
@vvakame is there a way for me to continue testing disabled definitions? Or do I need to manually rename them?

**Comment by vvakame** [comment]
> do I need to manually rename them?

yes. PR is welcome. :+1: but disabled file is huge file... (https://github.com/borisyankov/DefinitelyTyped/commit/3933f040ef054ea34531e756c7e7f31840afd38c

**Comment by vvakame** [comment]
@FranPregernik what revision of js-data.d.ts did you use?

**Comment by FranPregernik** [comment]
Hi,

I think I used this exact one  (e02dc63).

hope this helps also:
```
~/W/P/t/plm-mock git:master ❯❯❯ tsd --version
>> tsd 0.6.4

~/W/P/t/plm-mock git:master ❯❯❯ tsd query js-data --info
 - js-data / js-data    
   >> JSData 1.5.4     : github.com/js-data/js-data
    @ Stefan Steinhart : github.com/reppners

~/W/P/t/plm-mock git:master ❯❯❯ tsd query js-data --commit
 - js-data / js-data : e02dc6 : 2015-09-03 12:10
      
   4fb28c | 2015-02-25 21:45 | reppners   
    | + initial commit   
      
   143ee5 | 2015-02-26 08:53 | reppners   
    | + added tests   
    |    
    | + changed type definition to expose definitions as module and as globally defined namespace var   
      
   eec429 | 2015-03-02 20:27 | reppners   
    | + changing the JSData Promise type to interface   
    |    
    | + adding definitions object with usage example in tests for enabling custom type definitions of   
    | created resource definitions   
      
   5f2cc5 | 2015-03-04 08:09 | reppners   
    | + renaming module JSData_ to JSData since it is possible to declare var and module with the   
    | same name   
      
   a6cbfb | 2015-03-05 07:42 | reppners   
    | + changed promise declaration to own custom since it is not possible to reuse ES6 polyfill   
    | promise class declaration   
      
   e26a04 | 2015-03-06 07:23 | reppners   
    | + update of definitions to make use of union types   
      
   2ffc04 | 2015-08-21 23:27 | Daniel Rosenwasser @ DanielRosenwasser   
    | Fixed definitions for 'js-data'.   
      
   e02dc6 | 2015-09-03 12:10 | vvakame @ vvakame   
    | adhoc fix js-data/js-data.d.ts and js-data-http/js-data-http-tests.ts
```

**Comment by vvakame** [comment]
@FranPregernik ah, i see.
https://github.com/borisyankov/DefinitelyTyped/blob/master/js-data/js-data.d.ts#L156
this is [intersection types](https://github.com/Microsoft/TypeScript/pull/3622). this feature is new to 1.6.0-beta.
please use tsc 1.6.0-beta or older revision of js-data.d.ts.

**Pull request merged**